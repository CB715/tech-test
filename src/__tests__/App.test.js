import React from 'react';
import { render, screen } from '@testing-library/react';
import App from '../components/App';

describe('App', () => {

  it('renders App component', () => {
    const { asFragment } = render(<App />);

    expect(asFragment()).toMatchSnapshot();
  });

  it('renders the logo', () => {
    render(<App />);
    const nasaLogo = screen.getByTestId('nasaLogo');

    expect(nasaLogo).toBeInTheDocument();
    expect(nasaLogo.tagName).toBe('IMG');
    expect(nasaLogo.classList).toContain('nasa-logo');
  });

});