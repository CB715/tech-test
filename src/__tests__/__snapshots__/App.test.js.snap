// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders App component 1`] = `
<DocumentFragment>
  <div
    class="App"
  >
    <img
      alt="NASA Logo"
      class="nasa-logo"
      src="https://cdn.cnn.com/cnnnext/dam/assets/200424060716-nasa-worm-logo.jpg"
    />
    <input
      class="search-box"
      placeholder="Search"
      type="text"
    />
  </div>
</DocumentFragment>
`;
